src/core/model.jl

```diff
@@ -205,7 +205,7 @@ Return an iterator on the components in a model's model instance.
 
 Return a DatumDef for `item` in the given component `comp_def`.
 """
-function datumdef(comp_def::ComponentDef, item::Symbol)
+function datumdef(comp_def::AbstractComponentDef, item::Symbol)
```

Comment:
@ckingdon95 do you see a downside to doing it this way (generalizing to one signature for all `AbstractComponentDef` types  instead of making a separate method for a `CompositeComponentDef`?  All tests pass, but you have a better sense of the theory behind this so I want to make sure I'm not breaking something.